heat/core/dndarray.py

```diff
@@ -3479,6 +3479,33 @@ def tanh(self, out=None):
         """
         return trigonometrics.tanh(self, out)
 
+    def tolist(self):
+        """
+        Return a copy of the local array data as a (nested) Python list. For scalars, a standard Python number is returned.
+
+        Returns
+        -------
+        list: list
+            A copy of the array as a python list.
+
+        Examples
+        --------
+        >>> a = ht.array([[0,1],[2,3]])
+        >>> a.tolist()
+        [[0, 1], [2, 3]]
+
+        >>> a = ht.array([[0,1],[2,3]], split=0)
+        >>> a.tolist()
+        (1/2) [[0, 1]]
+        (2/2) [[2, 3]]
+
+        >>> a = ht.array([[0,1],[2,3]], split=1)
+        >>> a.tolist()
+        (1/2) [[0], [2]]
+        (2/2) [[1], [3]]
+        """
+        return self.__array.tolist()
```

Comment:
The question is what we are expecting here. Do we want to have the partial lists of each DNDarray part or a gathered version. If we want to be consistent, e.g. with numpy(), we actually want to have the resplit(None) behaviour here.

Reply:
I added a parameter keepsplit which defaults to False. Personally, however, I would only expect the full list if split==None, and a partial list otherwise.

Reply:
And this behaviour is implemented?

Reply:
Keepdim defaults to False, so it will first gather the whole array (this is what you suggested)

Reply:
Yep my bad, one of the code lines was hidden in the preview. All good